inc/rest-api/class-post-fields.php

```diff
@@ -251,6 +253,39 @@ private function get_schema( $children ) {
 	 * @return array
 	 */
 	private function get_block_editor_post_types() {
-		return array_filter( get_post_types(), 'use_block_editor_for_post_type' );
+		return array_filter(
+			\get_post_types(),
+			function( $post_type ) {
+				if ( ! post_type_exists( $post_type ) ) {
+					return false;
+				}
+
+				if ( ! post_type_supports( $post_type, 'editor' ) ) {
+					return false;
+				}
+
+				$post_type_object = get_post_type_object( $post_type );
+				if ( $post_type_object && ! $post_type_object->show_in_rest ) {
+					return false;
+				}
+				return true;
```

Comment:
You should _also_ apply `use_block_editor_for_post_type` here because it's further filterable at a theme or plugin level.